### PR TITLE
Make the Dockerfile build as root and do the user creation at the end

### PR DESCRIPTION
**Pull request by TrentHouliston**
This means that cases where the $USERID who is using the dockerfile is not 1000 can build a docker image with their own ID so you don't get conflicts

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -272,13 +262,24 @@ RUN install-package \
...
-#######################################
+##############################
+### SETUP THE USER ACCOUNT ###
+##############################
+
+# Setup the sudo so it can be run without a password
+COPY etc/sudoers.d/user /etc/sudoers.d/user
+RUN chmod 440 /etc/sudoers.d/user
+
+# Create the fourtel user, and setup sudo so no password is required
```
Comment (Bidski) [inline docker/Dockerfile:277]: Excuse me? Which user??

File: docker/Dockerfile [hunk 2]
```diff
@@ -272,13 +262,24 @@ RUN install-package \
...
 # Install formatters as last step so if they change the image is easy to update
 RUN pip install --upgrade \
-    cmake-format==0.6.9 \
+    cmake-format==0.6.10 \
     isort==4.3.21 \
     black==19.10b0
 
-#######################################
-### ADD NEW PROGRAMS/LIBRARIES HERE ###
-#######################################
```
Comment (Bidski) [inline docker/Dockerfile:281]: Add this back in

File: docker/Dockerfile (hunk 1 above)
Comment (TrentHouliston) [reply docker/Dockerfile:277]: This is why you don't put constants in comments :P

**Comment by JosephusPaye** [comment]
No looking forward to rebuilding my image 😬😅

File: cmake/Modules/ToolchainLibraryFinder.cmake [hunk 3]
```diff
@@ -108,13 +108,9 @@ function(ToolchainLibraryFinder)
...
-    FOUND_VAR
-    ${PACKAGE_NAME}_FOUND
-    REQUIRED_VARS
-    ${required_vars}
-    VERSION_VAR
-    ${PACKAGE_NAME}_VERSION
-    # VERSION_VAR "${MAJOR}.${MINOR}.${PATCH}")
+    FOUND_VAR ${PACKAGE_NAME}_FOUND
+    REQUIRED_VARS ${required_vars}
+    VERSION_VAR ${PACKAGE_NAME}_VERSION # VERSION_VAR "${MAJOR}.${MINOR}.${PATCH}")
```
Comment (Bidski) [inline cmake/Modules/ToolchainLibraryFinder.cmake:113]: Why is this here?

File: nuclear/cmake/Modules/HeaderLibrary.cmake [hunk 4]
```diff
@@ -58,12 +58,9 @@ function(HeaderLibrary)
...
     ${PACKAGE_NAME}
-    FOUND_VAR
-    ${PACKAGE_NAME}_FOUND
-    REQUIRED_VARS
-    ${required_vars}
-    VERSION_VAR
-    ${PACKAGE_NAME}_VERSION
+    FOUND_VAR ${PACKAGE_NAME}_FOUND
+    REQUIRED_VARS ${required_vars}
+    VERSION_VAR ${PACKAGE_NAME}_VERSION
```
Comment (Bidski) [inline nuclear/cmake/Modules/HeaderLibrary.cmake:63]: And this?

File: nuclear/cmake/Modules/ToolchainLibraryFinder.cmake [hunk 5]
```diff
@@ -108,13 +108,9 @@ function(ToolchainLibraryFinder)
...
-    FOUND_VAR
-    ${PACKAGE_NAME}_FOUND
-    REQUIRED_VARS
-    ${required_vars}
-    VERSION_VAR
-    ${PACKAGE_NAME}_VERSION
-    # VERSION_VAR "${MAJOR}.${MINOR}.${PATCH}")
+    FOUND_VAR ${PACKAGE_NAME}_FOUND
+    REQUIRED_VARS ${required_vars}
+    VERSION_VAR ${PACKAGE_NAME}_VERSION # VERSION_VAR "${MAJOR}.${MINOR}.${PATCH}")
```
Comment (Bidski) [inline nuclear/cmake/Modules/ToolchainLibraryFinder.cmake:113]: And this?

File: cmake/Modules/ToolchainLibraryFinder.cmake (hunk 3 above)
Comment (TrentHouliston) [reply cmake/Modules/ToolchainLibraryFinder.cmake:113]: Because it's what cmake_format demands

Comment (Bidski) [inline cmake/Modules/ToolchainLibraryFinder.cmake:113]:
```suggestion
    VERSION_VAR ${PACKAGE_NAME}_VERSION # VERSION_VAR "${MAJOR}.${MINOR}.${PATCH}"
```

File: nuclear/cmake/Modules/ToolchainLibraryFinder.cmake (hunk 5 above)
Comment (Bidski) [inline nuclear/cmake/Modules/ToolchainLibraryFinder.cmake:113]:
```suggestion
    VERSION_VAR ${PACKAGE_NAME}_VERSION # VERSION_VAR "${MAJOR}.${MINOR}.${PATCH}"
```
